lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java

```diff
@@ -1,31 +1,44 @@
 package org.enso.syntax2;
 
-import org.enso.syntax2.Message;
 import java.io.File;
+import java.net.URISyntaxException;
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
 import java.nio.charset.StandardCharsets;
 
 public final class Parser implements AutoCloseable {
   static {
     String os = System.getProperty("os.name");
-    File dir = new File(".").getAbsoluteFile();
-    for (; ; ) {
-      File parser;
-      if (os.startsWith("Mac")) {
-        parser = new File(dir, "target/rust/debug/libenso_parser.dylib");
-      } else if (os.startsWith("Windows")) {
-        parser = new File(dir, "target/rust/debug/enso_parser.dll");
-      } else {
-        parser = new File(dir, "target/rust/debug/libenso_parser.so");
-      }
-      try {
-        System.load(parser.getAbsolutePath());
-        break;
-      } catch (LinkageError e) {
-        dir = dir.getParentFile();
-        if (dir == null) {
-          throw e;
+    String name;
+    if (os.startsWith("Mac")) {
+      name = "libenso_parser.dylib";
+    } else if (os.startsWith("Windows")) {
+      name = "enso_parser.dll";
+    } else {
+      name = "libenso_parser.so";
+    }
+
+    File parser = null;
+    try {
+      var whereAmI = Parser.class.getProtectionDomain().getCodeSource().getLocation();
+      File dir = new File(whereAmI.toURI()).getParentFile();
+      parser = new File(dir, name);
+      System.load(parser.getAbsolutePath());
```

Comment:
When the parser is used in the `engine`, this is the line that loads the library into JVM.